sql/sql_servers.cc

```diff
@@ -166,6 +215,18 @@ bool servers_init(bool dont_read_servers_table)
     DBUG_RETURN(TRUE);
   thd->thread_stack= (char*) &thd;
   thd->store_globals();
+
+  // Check transaction_registry table schema.
+  if (trt_table_correct(thd, trt_correct))
```

Comment:
This has nothing to do with `mysql.servers` table. Place schema check into `mysqld_main()` after `Events::init()` (mysqld.cc:6024). Make your own temporary `thd`.